apps/console/src/features/notification-channels/settings/settings-section.tsx

```diff
@@ -0,0 +1,364 @@
+    placeholder?: string;
+    primaryAction?: any;
+    primaryActionDisabled?: boolean;
+    primaryActionIcon?: SemanticICONS;
+    secondaryAction?: any;
+    secondaryActionDisabled?: boolean;
+    secondaryActionIcon?: SemanticICONS;
+    showActionBar?: boolean;
+    topActionBar?: React.ReactNode;
+}
+
+/**
+ * Settings section component.
+ *
+ * @param props - Props injected to the component.
+ * @returns Settings section component.
+ */
+export const SettingsSection: FunctionComponent<PropsWithChildren<SettingsSectionProps>> = (
+    props: PropsWithChildren<SettingsSectionProps>
+): JSX.Element => {
+
+    const {
+        children,
+        className,
+        contentPadding,
+        description,
+        connectorEnabled,
+        header,
+        icon,
+        onPrimaryActionClick,
+        onSecondaryActionClick,
+        placeholder,
+        primaryAction,
+        primaryActionDisabled,
+        primaryActionIcon,
+        secondaryAction,
+        secondaryActionDisabled,
+        secondaryActionIcon,
+        showActionBar,
+        ["data-testid"]: testId,
+        topActionBar
+    } = props;
+
+    const { isGreaterThanOrEqualComputerViewport } = useMediaContext();
+
+    const classes: string = classNames({
+        "with-top-action-bar": topActionBar !== undefined
+    }, className);
+
+    /**
+     * Construct the action element.
+     *
+     * @param action - action which is passed in.
+     * @param actionIcon - Icon for the action.
+     * @param actionDisabled - Flag to determine if the action should be disabled.
+     * @param actionOnClick - On Click handler of the action.
+     * @param actionType - Type of the action.
+     * @returns Constructed element.
+     */
+    const constructAction = (
+        action: any,
+        actionIcon: SemanticICONS,
+        actionDisabled: boolean,
+        actionOnClick: any,
+        actionType: "primary" | "secondary"
+    ) => {
+        // if passed in action is a react component
+        if (typeof action === "function" || typeof action === "object") {
+            return (
+                <List.Content
+                    className={ actionDisabled ? "disabled" : "" }
+                    floated={ actionType === "secondary" ? "right" : "left" }
+                >
+                    { action }
+                </List.Content>
+            );
+        }
+
+        // if passed in action is of type `string`.
+        if (typeof action === "string") {
+            return (
+                <List.Content
+                    className={ actionDisabled ? "disabled" : "" }
+                    floated={ actionType === "secondary" ? "right" : "left" }
+                >
+                    <List.Header className="action-button-text" onClick={ actionOnClick }>
+                        {
+                            actionIcon
+                                ? (<><Icon name={ actionIcon }/>{ " " }</>)
```

Comment:
```suggestion
                                ? <Icon name={ actionIcon }/>)
```
We might be able to use a `margin` or a `padding` for spacing here with CSS